features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/EndpointsNew/EndpointOverview.jsx

```diff
@@ -0,0 +1,577 @@
+        borderRight: '#c4c4c4',
+        borderRightStyle: 'solid',
+        borderRightWidth: 'thin',
+    },
+    endpointsWrapperRight: {
+        padding: theme.spacing.unit,
+    },
+    endpointsTypeSelectWrapper: {
+        marginLeft: theme.spacing.unit * 2,
+        marginRight: theme.spacing.unit * 2,
+        padding: theme.spacing.unit,
+        display: 'flex',
+        justifyContent: 'space-between',
+    },
+    endpointTypesSelectWrapper: {
+        display: 'flex',
+    },
+    defaultEndpointWrapper: {
+        paddingLeft: theme.spacing.unit,
+        paddingRight: theme.spacing.unit,
+        marginRight: theme.spacing.unit,
+    },
+    configDialogHeader: {
+        fontWeight: '600',
+    },
+});
+
+/**
+ * The endpoint overview component. This component holds the views of endpoint creation and configuration.
+ * @param {any} props The props that are being passed to the component.
+ * @returns {any} HTML view of the endpoints overview.
+ */
+function EndpointOverview(props) {
+    const { classes, api, onChangeAPI } = props;
+    const { endpointConfig, endpointSecurity } = api;
+    const [endpointType, setEndpointType] = useState({ key: 'http', value: 'HTTP/REST Endpoint' });
+    const [epConfig, setEpConfig] = useState(endpointConfig);
+    const [endpointSecurityInfo, setEndpointSecurityInfo] = useState({});
+    const [isLBConfigOpen, setLBConfigOpen] = useState(false);
+    const [advanceConfigOptions, setAdvancedConfigOptions] = useState({
+        open: false,
+        index: 0,
+        type: '',
+        category: '',
+        config: undefined,
+    });
+
+    const endpointTypes = [{ key: 'http', value: 'HTTP/REST Endpoint' },
+        { key: 'address', value: 'HTTP/SOAP Endpoint' }];
+
+    /**
+     * Method to get the type of the endpoint. (HTTP/REST or HTTP/SOAP)
+     * In failover/ loadbalance cases, the endpoint type is presented in the endpoints list. Therefore that property
+     * needs to be extracted separately.
+     *
+     * @param {string} type  The representative type of the endpoint.
+     * @return {string} The type of the endpoint.
+     * */
+    const getEndpointType = (type) => {
+        if (type === 'http') {
+            return endpointTypes[0];
+        } else if (type === 'address') {
+            return endpointTypes[1];
+        } else {
+            const prodEndpoints = endpointConfig.production_endpoints;
+            if (Array.isArray(prodEndpoints)) {
+                return prodEndpoints[0].endpoint_type !== undefined ? endpointTypes[1] : endpointTypes[0];
+            }
+            return prodEndpoints.endpoint_type !== undefined ? endpointTypes[1] : endpointTypes[0];
+        }
+    };
+    useEffect(() => {
+        setEpConfig(endpointConfig);
+    }, []);
+    useEffect(() => {
+        setEndpointType(getEndpointType(endpointConfig.endpoint_type));
+    }, []);
+    useEffect(() => {
+        setEndpointSecurityInfo(endpointSecurity);
```

Comment:
we do not need 3 use effects here right? pre conditions for all 3 seems to be the same. I think we can move it to one useEffect

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/6725/commits/625591235cad25fe1786ea5547d1f85d34a75dd2